src/functions.php

```diff
@@ -61,7 +61,11 @@ function prepend($stream, $callback, $read_write = STREAM_FILTER_ALL)
 function fun($filter, $params = null)
 {
     $fp = fopen('php://memory', 'w');
-    $filter = @stream_filter_append($fp, $filter, STREAM_FILTER_WRITE, $params);
+    if (null === $params) {
+        $filter = @stream_filter_append($fp, $filter, STREAM_FILTER_WRITE);
```

Comment:
Internally, PHP will use a good default value for `$params` here. The PHP docs says `null` but it will automatically change to `[]` if the filter expects it.